aeron/logbuffer/term/reader.go

```diff
@@ -41,12 +48,15 @@ func Read(termBuffer *atomic.Buffer, termOffset int32, handler FragmentHandler,
 		termOffset += util.AlignInt32(frameLength, logbuffer.FrameAlignment)
 
 		if !logbuffer.IsPaddingFrame(termBuffer, fragmentOffset) {
+			fragmentsRead++
 			header.Wrap(termBuffer.Ptr(), termBuffer.Capacity())
 			header.SetOffset(fragmentOffset)
-			handler(termBuffer, fragmentOffset+logbuffer.DataFrameHeader.Length,
+			err := handler(termBuffer, fragmentOffset+logbuffer.DataFrameHeader.Length,
 				frameLength-logbuffer.DataFrameHeader.Length, header)
-
-			fragmentsRead++
+			if err != nil {
```

Comment:
I wonder if we should return the error here instead of calling the error handler to make the api more consistent?